### PR TITLE
minizinc, minizincide: 2.4.3 -> 2.5.3

### PR DESCRIPTION
**Pull request by esclear**
###### Motivation for this change
This PR updates minizinc and minizincide from versions 2.4.3 to 2.5.3 respectively.

The patches are no longer necessary as of https://github.com/MiniZinc/libminizinc/commit/952ffda0bd23dc21f83d3e3f080ea5b3a414e8e0.

For the IDE, `mkDerivation` is used so that Qt works and submodules must be fetched, since there is a new submodule that needs to be feched (see https://github.com/MiniZinc/MiniZincIDE/commit/9a8bb28f7d6784581f9da96a1355de3b23bacaa0).

This is related to #104331.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 112514` at ba5d7398 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/c81adad590df6d077e7e8d8450efb147">minizinc</a></li></li>
    <li><a href="https://gist.github.com/a79789dd3b077ee3e2a8b705b40f3d56">minizincide</a></li></li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 112514` at ba5d7398 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/4de75c317768d7584ed0308c8fb69f14">minizinc</a></li></li>
    <li><a href="https://gist.github.com/cc58a71b2016ab74fd4d259f14ea313d">minizincide</a></li></li>
  </ul>
</details>

File: pkgs/development/tools/minizinc/default.nix [hunk 1]
```diff
@@ -12,28 +12,9 @@ stdenv.mkDerivation {
...
     repo = "libminizinc";
     rev = version;
-    sha256 = "0mahf621zwwywimly5nd6j39j7qr48k5p7zwpfqnjq4wn010mbf8";
+    sha256 = "1kc65sxkc64pr560qaaznc44jnlvq7pbpzwijad410lpcnna5byg";
   };
 
-  patches = [
-    # Fix build with newer Bison versions:
-    # https://github.com/MiniZinc/libminizinc/issues/389
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/development/tools/minizinc/default.nix:21]: Please remove fetchpatch from inputs.

File: pkgs/development/tools/minizinc/default.nix [hunk 2]
```diff
@@ -1,6 +1,6 @@
 { lib, stdenv, fetchFromGitHub, fetchpatch, cmake, flex, bison }:
 let
-  version = "2.4.3";
+  version = "2.5.3";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/minizinc/default.nix:3]: That let in could be converted into a rec.

File: pkgs/development/tools/minizinc/ide.nix [hunk 3]
```diff
@@ -1,19 +1,20 @@
-{ lib, stdenv, fetchFromGitHub, qtbase, qtwebengine, qtwebkit, qmake, makeWrapper, minizinc }:
+{ lib, mkDerivation, fetchFromGitHub, qtbase, qtwebengine, qtwebkit, qmake, minizinc }:
 let
-  version = "2.4.3";
+  version = "2.5.3";
 in
-stdenv.mkDerivation {
+mkDerivation {
```
Comment (SuperSandro2000) [inline pkgs/development/tools/minizinc/ide.nix:5]: I am not sure if this breaks something.

Comment (esclear) [reply pkgs/development/tools/minizinc/ide.nix:5]: This change is what was required to make it work with Qt at all.

Comment (SuperSandro2000) [reply pkgs/development/tools/minizinc/ide.nix:5]: Oh, yeah. It is required for QT.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112514` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>minizinc</li>
</ul>
</details>

**Pull request merged**